tests/Core/Layer/Collector/TagValueRegexCollectorTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Qossmic\Deptrac\Core\Layer\Collector;
+
+use PHPUnit\Framework\TestCase;
+use Qossmic\Deptrac\Contract\Config\Collector\TagValueRegexConfig;
+use Qossmic\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
+use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeReference;
+use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeToken;
+use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeType;
+use Qossmic\Deptrac\Core\Layer\Collector\TagValueRegexCollector;
+
+final class TagValueRegexCollectorTest extends TestCase
+{
+    private TagValueRegexCollector $collector;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->collector = new TagValueRegexCollector();
+    }
+
+    public static function dataProviderSatisfy(): iterable
+    {
+        yield [TagValueRegexConfig::create('@foo'), ['@foo' => ['']]];
```

Comment:
Being nitpicky:
The other provider also set a key for the yield, it would be nice to have them here to.

Reply:
done